src/join.rs

```diff
@@ -144,51 +134,33 @@ impl<'a, T> Get for GetMut<'a, T>
         let x: &mut Self = std::mem::transmute(self);
         x.0.get_mut(idx)
     }
-}
+}*/
 
 /// Joined is an Iterator over a group of `Storages`
-pub struct Joined<K, V> {
-    keys: bitset::Iter<K>,
-    values: V,
+pub struct Joined<O: Open> {
+    keys: BitIter<O::Mask>,
+    values: O::Value,
 }
 
-impl<K, V> Joined<K, V>
-    where K: BitSetLike
-{
-    fn new(k: K, v: V) -> Joined<K, V> {
+impl<O: Open> From<O> for Joined<O> {
+    fn from(o: O) -> Self {
+        let (keys, values) = o.open();
         Joined {
-            keys: k.iter(),
-            values: v,
+            keys: keys.iter(),
+            values: values,
         }
     }
 }
 
-impl<K, V> std::iter::Iterator for Joined<K, V>
-    where K: BitSetLike,
-          V: Get
-{
-    type Item = V::Value;
-    fn next(&mut self) -> Option<Self::Item> {
-        if let Some(idx) = self.keys.next() {
-            unsafe { Some(self.values.get(idx)) }
-        } else {
-            None
-        }
+impl<O: Open> std::iter::Iterator for Joined<O> {
+    type Item = O::Type;
+    fn next(&mut self) -> Option<O::Type> {
+        self.keys.next().map(|idx| unsafe {
+            O::get(self.values, idx)
```

Comment:
I believe I have to do the mutable transmute there, similarly to the original code...